### PR TITLE
Merge `bugzilla.py` module with `models.py`

### PR DESCRIPTION
**Pull request by grahamalama**
Since `bugzilla.py` was a collection of Pydantic `BaseModel`s, in this PR we move them into `models.py` so that it is the one source of all models used in the app.

### CODE REVIEW

File: jbi/services.py [hunk 1]
```diff
@@ -1,15 +1,19 @@
 """Services and functions that can be used to create custom actions"""
+from __future__ import annotations
```
Comment (bsieber-mozilla) [inline jbi/services.py:2]: What is`annotations` used for?

File: jbi/services.py [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 import backoff
 import bugzilla as rh_bugzilla
 from atlassian import Jira, errors
 from statsd.defaults.env import statsd
 
 from jbi import environment
-from jbi.models import Actions
+
+if TYPE_CHECKING:
+    from jbi.models import Actions
```
Comment (bsieber-mozilla) [inline jbi/services.py:16]: Only import on `TYPE_CHECKING`? (What value is TYPE_CHECKING? are we expecting this as False?)

**Review by bsieber-mozilla — APPROVED** [review]
Minor resolvable comments, overall lgtm!

**Comment by grahamalama** [comment]
@bsieber-mozilla 
> What is annotations used for?

> Only import on TYPE_CHECKING? (What value is TYPE_CHECKING? are we expecting this as False?)

I was running into a circular import after making this change, and the only reason for that was for the purposes of type hinting. I found [this](https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/) blog post that pointed me to add the changes that you called out in your review.

**Pull request merged**